hooks/publish.py

```diff
@@ -426,52 +453,49 @@ def publish(self, task_settings, item):
         publisher = self.parent
 
         # Get item properties populated by validate method
-        publish_name          = item.properties.publish_name
-        publish_path          = item.properties.publish_path
-        publish_symlink_path  = item.properties.publish_symlink_path
-        publish_type          = item.properties.publish_type
-        publish_version       = item.properties.publish_version
+        publish_name          = item.get_property("publish_name")
+        publish_path          = item.get_property("publish_path")
+        publish_symlink_path  = item.get_property("publish_symlink_path")
+        publish_type          = item.get_property("publish_type")
+        publish_version       = item.get_property("publish_version")
         publish_user          = item.get_property("publish_user", default_value=None)
 
+        task_publish_paths = item.get_property("publish_paths_expanded", [])
+
         # handle publishing of files first
-        publish_files = self.publish_files(task_settings, item, publish_path)
-        if "publish_files" not in item.properties:
-            item.properties.publish_files = {}
-        item.properties.publish_files[self.name] = publish_files
+        task_publish_paths.append(self.publish_files(task_settings, item, publish_path))
 
         # symlink the files if it's defined in publish templates
-        publish_symlinks = []
         if publish_symlink_path:
-            publish_symlinks = self.symlink_publishes(
+            task_publish_paths.append(self.symlink_publishes(
                 task_settings,
                 item,
                 publish_path,
                 publish_symlink_path
-            )
-        if "publish_symlinks" not in item.properties:
-            item.properties.publish_symlinks = {}
-        item.properties.publish_symlinks[self.name] = publish_symlinks
+            ))
+
+        item.local_properties["publish_paths_expanded"] = task_publish_paths
 
         # Get any upstream dependency paths
-        dependency_paths = item.properties.get("publish_dependency_paths", [])
+        dependency_paths = self._get_dependency_paths(task_settings, item)
 
         # Get any upstream dependency ids
-        dependency_ids = item.properties.get("publish_dependency_ids", [])
+        dependency_ids = self._get_dependency_ids(task_settings, item)
```

Comment:
Have you confirmed this isn't getting called multiple times? Perhaps we should remove the other places where this gets called/stored?

Reply:
Nevermind, I see the method overrides at the bottom.